src/debuginfo.cpp

```diff
@@ -811,14 +831,17 @@ void jl_getFunctionInfo(const char **name, size_t *line, const char **filename,
         if ((*it).second.func) {
             DISubprogram debugscope =
                 DISubprogram(prev.Loc.getScope((*it).second.func->getContext()));
-            *filename = debugscope.getFilename().data();
+            jl_copy_str(filename, debugscope.getFilename().data());
```

Comment:
> Well, IIUC the assumption was that these did not need to be freed because the owning objects are live for the lifetime of the program

It sure is given we've never seen any crash related to it (or do we ;-p ).

Another solution (to solve the issue that the string we assign to sometimes needs to be free'd and sometimes doesn't) is to record whether it needs to be free'd along with the string. However, that will make the code much more complicated (especially in C) and this is not performance critical code anyway....

> We seem to have gotten away with it so far.

I don't disagree that we should probably fix this. I'd prefer not to do it in this PR since I don't want to make this harder to review because of an issue that might not happen in practice.